src/php/ext/grpc/channel.c

```diff
@@ -162,6 +166,72 @@ void generate_sha1_str(char *sha1str, char *str, php_grpc_int len) {
   make_sha1_digest(sha1str, digest);
 }
 
+bool php_grpc_persistent_list_delete_unused_channel(char* target, target_bound_le_t* target_bound_status){
+  char *key = NULL;
+  zval *data;
+  int key_type;
+  PHP_GRPC_HASH_FOREACH_STR_KEY_VAL_START(&grpc_persistent_list, key, key_type, data)
+    if (key_type != HASH_KEY_IS_STRING || key == NULL) {
+      zend_throw_exception(spl_ce_InvalidArgumentException,
+                           "args keys must be strings", 1 TSRMLS_CC);
+      return FAILURE;
+    }
+    php_grpc_zend_resource *rsrc  = (php_grpc_zend_resource*) PHP_GRPC_HASH_VALPTR_TO_VAL(data)
+    if (rsrc == NULL) {
+      break;
+    }
+    channel_persistent_le_t* le = rsrc->ptr;
+    // Find the channel sharing the same target.
+    if (strcmp(le->channel->target, target) == 0) {
+      // ref_count=1 means that only the map holds the reference to the channel.
+      if(le->channel->ref_count == 1) {
+        php_grpc_delete_persistent_list_entry(le->channel->key,
```

Comment:
Unref the channel?

Reply:
Inside `php_grpc_delete_persistent_list_entry`, it will call  [php_grpc_zend_hash_del](https://github.com/grpc/grpc/pull/15218/files/591fd4657130e335e0ae91747b2190498752a3e7#diff-f942bdda09aac4a837cef783506850f0R593), which will call the handler [`php_grpc_channel_plink_dtor`](https://github.com/grpc/grpc/pull/15218/files/591fd4657130e335e0ae91747b2190498752a3e7#diff-f942bdda09aac4a837cef783506850f0R599) remove the map element.
The `unref` is called there.